src/safeposix/syscalls/net_calls.rs

```diff
@@ -1765,22 +1765,103 @@ impl Cage {
+    ///
+    /// * EMFILE - The per-process limit on the number of open file
+    ///            descriptors has been reached.
+    ///
+    /// * ENFILE - The system-wide limit on the total number of open files
+    ///            has been reached.
+    ///
+    /// * ENOMEM - Not enough free memory.  This often means that the memory
+    ///            allocation is limited by the socket buffer limits, not by
+    ///            the system memory.
+    ///
+    /// ** ENOTSOCK - The file descriptor sockfd does not refer to a socket.
+    ///
+    /// ** EOPNOTSUPP - The referenced socket is not of type SOCK_STREAM.
+    ///
+    /// * EPERM - Firewall rules forbid connection.
+    ///
+    /// * EPROTO - Protocol error.
+    ///
+    /// In addition, network errors for the new socket and as defined for
+    /// the protocol may be returned.  Various Linux kernels can return
+    /// other errors such as ENOSR, ESOCKTNOSUPPORT, EPROTONOSUPPORT,
+    /// ETIMEDOUT.  The value ERESTARTSYS may be seen during a trace.
+    ///
+    /// ** Indicates the error may be returned from RustPOSIX
+    ///
+    /// ### Panics
+    ///
+    /// * invalid or out-of-bounds file descriptor), calling unwrap() on it will cause a panic.
+    /// * Unknown errno value from fcntl returned, will cause panic.
+    ///
+    /// for more detailed description of all the commands and return values, see
+    /// [accept(2)](https://linux.die.net/man/2/accept)
     pub fn accept_syscall(&self, fd: i32, addr: &mut interface::GenSockaddr) -> i32 {
+        //If fd is out of range of [0,MAXFD], process will panic
+        //Otherwise, we obtain a write gaurd to the Option<FileDescriptor> object
         let checkedfd = self.get_filedescriptor(fd).unwrap();
         let mut unlocked_fd = checkedfd.write();
         if let Some(filedesc_enum) = &mut *unlocked_fd {
+            //Find the next available file descriptor and grab a mutable reference
+            //to the Option<FileDescriptor> object
             let (newfd, guardopt) = self.get_next_fd(None);
+            //In the case that no fd is available from the call to get_next_fd,
+            //fd is set to -ENFILE = -23 and the error is propagated forward
             if newfd < 0 {
                 return fd;
             }
             let newfdoption: &mut Option<FileDescriptor> = &mut *guardopt.unwrap();
 
+            //Pattern match such that FileDescriptor object must be the Socket variant
+            //Otherwise, return with an err as the fd refers to something other than a socket
             match filedesc_enum {
                 Socket(ref mut sockfdobj) => {
+                    //Clone the socket handle as it may be in use by other threads and
+                    //obtain a read lock, blocking the calling thread until
+                    //there are no other writers that hold the lock
                     let sock_tmp = sockfdobj.handle.clone();
                     let mut sockhandle = sock_tmp.read();
```

Comment:
Does it have to be a mutable read guard? I deleted the mut keyword and all the tests still passed. 

(Thank you Qianxi for pointing it out)